src/FileHandler.F90

```diff
@@ -255,7 +255,7 @@ logical function is_lockable(this)
       class (FileHandler), intent(in) :: this
 
       is_lockable = .false.
-      if (associated(this%lock)) is_lockable = this%lock%is_initialized()
+      if (allocated(this%lock)) is_lockable = this%lock%is_initialized()
```

Comment:
Why is the check on initialized() necessary?   It was not before.

Reply:
You can allocate a lock without initialization. So it is wise to check. I don't know when you switch to this lazy initialization.

Reply:
ok - I'll provisionally accept it then.